src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/PhysicalAddress.cs

```diff
@@ -68,11 +68,6 @@ public override bool Equals(object comparand)
                 return false;
             }
 
-            if (_address.GetHashCode() != address._address.GetHashCode())
-            {
-                return false;
```

Comment:
Probably not. For that matter, I'm not sure the memoisation of the hash itself is worth doing. But maybe it really was found to be a hot path at some point.